http/src/main/java/io/micronaut/http/filter/FilterRunner.java

```diff
@@ -580,7 +592,12 @@ private ExecutionFlow<FilterContext> filter(FilterContext filterContext,
                 } else {
                     returnValue = method.invoke(bean, args);
                 }
-                return returnHandler.handle(filterContext, returnValue, methodContext.continuation);
+                ExecutionFlow<FilterContext> executionFlow = returnHandler.handle(filterContext, returnValue, methodContext.continuation);
+                PropagatedContext mutatedPropagatedContext = methodContext.mutablePropagatedContext.getContext();
```

Comment:
it would be nice if the return handler could do this, only if the filter actually defines a mutable context param. but probably overkill for this pr.